File: domain/src/main/java/com/arturogutierrez/openticator/domain/issuer/repository/IssuerRepository.java
package com.arturogutierrez.openticator.domain.issuer.repository;

import com.arturogutierrez.openticator.domain.issuer.model.Issuer;
import java.util.List;
import rx.Observable;

public interface IssuerRepository {

  Observable<List<Issuer>> getIssuers();
}

File: domain/src/main/kotlin/com/arturogutierrez/openticator/domain/issuer/repository/IssuerRepository.kt
package com.arturogutierrez.openticator.domain.issuer.repository

import com.arturogutierrez.openticator.domain.issuer.model.Issuer
import io.reactivex.Single
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class IssuerRepository @Inject constructor() {

  val issuers: Single<List<Issuer>>
    get() {
      val allIssuers = Issuer.values().asList().filter { it != Issuer.UNKNOWN }
      return Single.just(allIssuers)
    }
}
